### DISCUSSION TITLE
Speed based on Travel

### DISCUSSION THREAD

**Discussion by lunartester** [supermerill/SuperSlicer#2003] [Q&A]
How trivial (hours, days, weeks) it would be to implement a feature that would allow the end user to set the travel speed based on distance with at least 3 connected ranges.

For example an algorithm would automatically slice a part based on the range parameters all set by the end user.

If travel distance is set 0-5mm range then  I can also the travel speed at 40mm/sec

second range I can set the travel distance to 5mm-20mm then I can set to travel speed to 70mm/sec

the last range is anything above 20mm distance travel speed is set to 100mm/sec

Basically three concurrently connected ranges should cover most printers up to 400mm travel distance.

Cheers

**Comment by supermerill** [comment]
For me, something like an hour.
For someone that doesn't know the codebase, maybe 1 day.

But the most difficult thing to do isn't implementing it but to think about why and how.

Does it really address a problem?
Is it the best way to resolve it?
Does this doesn't over-complexify the application?
Does it have any side-effects for other people?

**Reply by lunartester** [reply]
This feature is not a substitute to stock option but an option vs. default option, it deals specifically with flexible TPU only and stringing of TPU. If this was an option not a substitute for default setting, how long it would take you to implement it? It needs another setting to be altered which is "wipe" without retraction so to set "wipe" I can choose not to use retraction at all. 

So basically this way "wipe" distance is more like (Coasting in Cura) so the Slicer would treat "wipe" as if it's another travel move going over existing tool-path but not extruding anymore, so if I can set that  "wipe" distance to any distance and tie it up with my 3 -range travel speed based on distance, this would solve an issue where you have large parts with small details in them and these small details are 2-3 mm a part, so using one travel speed would never work. 

In general this solves the issue of printing small parts that are very close to each other under 5mm and we have many of them spread out all over the built plate, so if i want to print at much faster speed then 20mm/sec retraction takes too much time on such small adn closely spaced parts even if it's 0.1mm retraction just the time it takes to reverse the extruder direction back and forth already creates stringing, so instead of retraction I simply use the tool-path to get rid of the extra filament over wipe distance. And if I have parts that are spaced under 5mm then I do not need fast travel plus there is not enough distance there to get to 100mm/sec travel speed, but if I have another part 100mm spaced a part I need to get to to it so fast that there is no oozing from the nozzle during travel when I get there. I've tested this theory as far as when you have fast travel on parts spaced 100mm it can get there fast enough without oozing, the issue is fine tuning things for shortly spaced parts whre we need wipe to work independently from retraction and ability to set slower speed on travel on short moves.

**Reply by supermerill** [reply]
speed is set at Gcode.cpp:l3947 (_compute_speed_mm_per_sec)
wipe is set at the last extrusion.
retract compute is set at (l4467) and called by travel_to (l4254), that can also modify (erase) the wipe.
all that happens/ is called in _before_extrude (l4026)

**Reply by lunartester** [reply]
I did not get understand answer, I'm not a programmer, did you specify where to alter your slicer code to make these changes or does your reply mean something else.
